src/fsharp/TypedTreeOps.fs

```diff
@@ -3183,6 +3183,28 @@ let TyconRefHasAttribute g m attribSpec tcref =
                     (fun _ -> Some ())
         |> Option.isSome
 
+/// Check if a type definition has an attribute with a specific full name
+let TyconRefHasAttributeByName (m: range) attrFullName (tcref: TyconRef) = 
```

Comment:
At first I thought this would be expensive every time `isByRefLikeTyconRef` is called, but we cache the result once the type has been checked the first time via: `tcref.SetIsByRefLike`